### ISSUE TITLE
function name can be built in names, but not overrides 

### ISSUE THREAD

**Issue by ThakeeNathees** [godotengine/godot#37024]
**Godot version:** 3.2.1

**OS/device including version:** Any

**Issue description:**
any function name can be built in names (built in func, const), but when calling it only subclass methods are valid. 
1. we have to make built in methods also overrideable (like python)
2. or prevent function names shadows built in.

**Steps to reproduce:**
![built-in-override](https://user-images.githubusercontent.com/41085900/76631246-5d535280-6567-11ea-928d-1b6fb14318a9.JPG)

**Minimal reproduction project:**
N/A

**Comment by dalexeev** [comment]
It seems that using `self` completely solves this problem.

```gdscript
func min():
    return "Hello"

func _run():
    print(min(1, 2)) # prints '1'
    print(self.min()) # prints 'Hello'
```

P.S. By the way, why is `@GDScript.min()` not a 'vararg' function?

**Comment by dreamsComeTrue** [comment]
Maybe other way around this could be global scope resolution operator, like in C++ (though it looks bit ugly and is not Pythonic-like)

```c++
::min()
```
??

**Comment by dalexeev** [comment]
@dreamsComeTrue I think it is not necessary. There are not so many global objects/functions, all of them are highlighted in a different color. Another operator can only confuse beginners. I think it’s pretty logical that built-in functions have the highest priority. Plus, there is always the choice to give custom objects a different name.

**Comment by dreamsComeTrue** [comment]
So maybe dropping a line in official docs about prefixing those names with 'self' operator would be enough to close it?

**Comment by dalexeev** [comment]
Yes, that's a good idea. We need to document this use case of `self`.

Here is an example with `load()` (I took the file from my project and reduced it as much as possible).

```gdscript
extends ConfigFile

const PATH = "user://config.cfg"

func _init():
    self.load()

func load(path = PATH):
    return .load(path)
```
